src/test/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusServiceSpec.scala

```diff
@@ -350,13 +380,261 @@ class ConsignmentStatusServiceSpec extends AnyFlatSpec with MockitoSugar with Re
+      consignmentIdCaptor.getValue should equal(expectedConsignmentId)
+      statusTypeCaptor.getValue should equal(nonUploadStatusType)
+      statusValueCaptor.getValue should equal(expectedStatusValue)
+      modifiedTimeCaptor.getValue should equal(modifiedTime)
+    }
+  }
+
+  forAll(statusTypes.filter(_ != "Upload")) { nonUploadStatusType =>
+    "updateConsignmentStatus" should s"throw an InputDataException when only a consignmentId and a non-'Upload' statusType of ${nonUploadStatusType} has been passed in" in {
+      val fixedUUIDSource = new FixedUUIDSource()
+      val expectedConsignmentId = fixedUUIDSource.uuid
+
+      val updateConsignmentStatusInput =
+        ConsignmentStatusInput(expectedConsignmentId, nonUploadStatusType, None)
+
+      val thrownException = intercept[Exception] {
+        consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+      }
+
+      thrownException.getMessage should equal(
+        s"Invalid ConsignmentStatus input: either '${nonUploadStatusType}' or ''"
+      )
+    }
+  }
+
+  "updateConsignmentStatus" should s"generate a 'Completed' statusValue when only a consignmentId and 'Upload' statusType have been passed in " +
+    "and all files have an 'Upload' statusValue of 'Success'" in {
+    val fixedUUIDSource = new FixedUUIDSource()
+    val expectedConsignmentId = fixedUUIDSource.uuid
+    val expectedStatusType = "Upload"
+    val expectedStatusValue = "Completed"
+    val statusValueCaptor: ArgumentCaptor[String] = ArgumentCaptor.forClass(classOf[String])
+
+    val consignmentStatusMockRepoResponse: Future[Int] = Future.successful(1)
+    val fileStatusMockRepoResponse: Future[Seq[FilestatusRow]] = Future.successful(
+      Seq(
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now()))
+      )
+    )
+    when(
+      consignmentStatusRepositoryMock.updateConsignmentStatus(
+        any[UUID],
+        any[String],
+        statusValueCaptor.capture(),
+        any[Timestamp]
+      )
+    ).thenReturn(consignmentStatusMockRepoResponse)
+
+    when(
+      fileStatusRepositoryMock.getFileStatus(
+        any[UUID],
+        any[String],
+        any[Option[Set[UUID]]]
+      )
+    ).thenReturn(fileStatusMockRepoResponse)
+
+    val updateConsignmentStatusInput =
+      ConsignmentStatusInput(expectedConsignmentId, expectedStatusType, None)
+
+    consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+
+    statusValueCaptor.getValue should equal(expectedStatusValue)
+  }
+
+  "updateConsignmentStatus" should s"generate a 'CompletedWithIssues' statusValue when only a consignmentId and 'Upload' statusType have been passed in " +
```

Comment:
Maybe re-word test title slightly to: `... when a consignmentId and 'Upload' ... `